pkg/snowflake/task.go

```diff
@@ -0,0 +1,345 @@
+func (tb *TaskBuilder) RemoveSessionParameters(params map[string]interface{}) string {
+	sortedKeys := make([]string, 0)
+	for k := range params {
+		sortedKeys = append(sortedKeys, k)
+	}
+	sort.Strings(sortedKeys)
+
+	return fmt.Sprintf(`ALTER TASK %v UNSET %v`, tb.QualifiedName(), strings.Join(sortedKeys, ", "))
+}
+
+// ChangeCondition returns the sql that will update the when condition for the task.
+func (tb *TaskBuilder) ChangeCondition(newCondition string) string {
+	return fmt.Sprintf(`ALTER TASK %v MODIFY WHEN %v`, tb.QualifiedName(), newCondition)
+}
+
+// ChangeSqlStatement returns the sql that will update the sql the task executes.
+func (tb *TaskBuilder) ChangeSqlStatement(newStatement string) string {
+	return fmt.Sprintf(`ALTER TASK %v MODIFY AS %v`, tb.QualifiedName(), EscapeString(newStatement))
+}
+
+// Suspend returns the sql that will suspend the task.
+func (tb *TaskBuilder) Suspend() string {
+	return fmt.Sprintf(`ALTER TASK %v SUSPEND`, tb.QualifiedName())
+}
+
+// Resume returns the sql that will resume the task.
+func (tb *TaskBuilder) Resume() string {
+	return fmt.Sprintf(`ALTER TASK %v RESUME`, tb.QualifiedName())
+}
+
+// Drop returns the sql that will remove the task.
+func (tb *TaskBuilder) Drop() string {
+	return fmt.Sprintf(`DROP TASK %v`, tb.QualifiedName())
+}
+
+// Describe returns the sql that will describe a task.
+func (tb *TaskBuilder) Describe() string {
+	return fmt.Sprintf(`DESCRIBE TASK %v`, tb.QualifiedName())
+}
+
+// Show returns the sql that will show a task.
+func (tb *TaskBuilder) Show() string {
+	return fmt.Sprintf(`SHOW TASKS LIKE '%v' IN DATABASE "%v"`, EscapeString(tb.name), EscapeString(tb.db))
+}
+
+// ShowParameters returns the query to show the session parameters for the task
+func (tb *TaskBuilder) ShowParameters() string {
+	return fmt.Sprintf(`SHOW PARAMETERS IN TASK %v`, tb.QualifiedName())
+}
+
+// SetDisabled disables the task builder
+func (tb *TaskBuilder) SetDisabled() *TaskBuilder {
+	tb.disabled = true
+	return tb
+}
+
+// IsDisabled returns if the task builder is disabled
+func (tb *TaskBuilder) IsDisabled() bool {
+	return tb.disabled
+}
+
+type task struct {
+	Id           string  `db:"id"`
+	CreatedOn    string  `db:"created_on"`
+	Name         string  `db:"name"`
+	DatabaseName string  `db:"database_name"`
+	SchemaName   string  `db:"schema_name"`
+	Owner        string  `db:"owner"`
+	Comment      *string `db:"comment"`
+	Warehouse    string  `db:"warehouse"`
+	Schedule     *string `db:"schedule"`
+	Predecessors *string `db:"predecessors"`
+	State        string  `db:"state"`
+	Definition   string  `db:"definition"`
+	Condition    *string `db:"condition"`
+}
+
+func (t *task) IsEnabled() bool {
+	return strings.ToLower(t.State) == "started"
+}
+
+func (t *task) GetPredecessorName() string {
+	if t.Predecessors == nil {
+		return ""
+	}
+
+	pre := strings.Split(*t.Predecessors, ".")
+	name, _ := strconv.Unquote(pre[len(pre)-1])
```

Comment:
Snowflake doesn't always wrap the name of the task in quotes and if there aren't quotes around the string then this errors and the result returned is an empty string

Reply:
```suggestion
		name, err := strconv.Unquote(pre[len(pre)-1])
	if err != nil {
		return pre[len(pre)-1]
	}

```

Reply:
If you modify the root task's name to be all caps it will reproduce in the acceptance tests